Qt6 fixes

bluebrother opened the pull request:
Replace obsolete classes and functions with their replacements, making it compile with current Qt6 preview.

bluebrother committed:
Use QElapsedTimer to measure time differences.

QTime::elapsed() is deprecated in recent Qt5, and removed in Qt6. Use
QElapsedTimer instead, which is the suggested replacement.

bluebrother committed:
Use QString::vasprintf() in newer Qt versions.

vsprintf() is deprecated in recent Qt5, and removed in Qt6. Its
replacement vasprintf() is available since Qt 5.5.

bluebrother committed:
Replace QRegExp with QRegularExpression.

QRegExp is deprecated in recent Qt, and has been removed in Qt6. Use the
replacement QRegularExpression.

bluebrother committed:
Replace QDateTime::fromTime_t for newer Qt.

fromTime_t has been deprecated and removed from Qt6. Use the replacement
fromSecsSinceEpoch() on Qt 5.8+.

bluebrother committed:
Use range iterator for creating QSet on newer Qt.

fromList() is deprecated and has been removed in Qt6. Since Qt 5.14
range constructors are available, use those instead.

cyberbobs commented:
Hello, thank you for the contribution. It will take some time to review and check (current test coverage is lacking, so there will be some serious manual checking). I'll try to get my hands on this next week. Will keep you posted.

cyberbobs committed:
FileAppender: don't force flush data to file system (behavior change!)

FileAppender will no longer call QFile::flush on every message write
due to it dramatically slowing apps when writing a lot of log records
to slow file system. The downside is some log data may be lost if
application crashes before QFile manages to write all of the buffered
data to file system.

It is still possible to revert to an old behavior calling
setFlushOnWrite(true) on FileAppender.

cyberbobs committed:
Updated to newer version of CMake

CMake started showing deprecation warning about CMake 2.8 files
support.

bluebrother committed:
Remove unused include.

Qt6 moves QTextCodec to the Qt5Compat module. Since it's unused anyway
remove it so we don't get the dependency on Qt5Compat.

bluebrother committed:
Fix destructor segfault with newer Qt versions.

Getting the values from a QMultiMap() returns a list, but not the same
on consecutive calls. Thus calling it twice to get the iterators will
yield different lists, causing the iterator to fail.

Fixes segfault when destructing the logger.
